File: packages/tauri-plugin-jarvis/guest-js/commands/fs.ts
import { invoke } from '@tauri-apps/api/core'

export function pathExists(path: string) {
  return invoke<boolean>('plugin:jarvis|path_exists', { path })
}

/**
 * This command is built into Jarvis App
 * Used to decompress a tarball file
 * @param path
 * @param destinationFolder
 * @param options
 * @returns
 */
export function decompressTarball(
  path: string,
  destinationFolder: string,
  options?: {
    overwrite?: boolean
  },
) {
  return invoke<string>('plugin:jarvis|decompress_tarball', {
    path,
    destinationFolder,
    overwrite: options?.overwrite ?? false,
  })
}

/**
 * Compress a given directory into a tarball file
 * @param srcDir Directory to compress
 * @param destFile destination file, should end with .tar.gz or .tgz
 * @param options
 * @returns
 */
export function compressTarball(
  srcDir: string,
  destFile: string,
  options?: {
    overwrite?: boolean
  },
) {
  return invoke<string>('plugin:jarvis|compress_tarball', {
    srcDir,
    destFile,
    overwrite: options?.overwrite ?? false,
  })
}

File: packages/tauri-plugin-jarvis/src/commands/fs.rs
use std::path::PathBuf;

use crate::utils;

#[tauri::command]
pub async fn path_exists(path: String) -> Result<bool, String> {
    Ok(std::path::Path::new(&path).exists())
}

#[tauri::command]
pub async fn decompress_tarball(
    path: PathBuf,
    destination_folder: PathBuf,
    overwrite: bool,
) -> Result<PathBuf, String> {
    utils::fs::decompress_tarball(path, destination_folder, overwrite).map_err(|e| e.to_string())
}

/// Compress a directory into a tarball
/// Both `src_dir` and `dest_file` must be be absolute paths
#[tauri::command]
pub async fn compress_tarball(
    src_dir: PathBuf,
    dest_file: PathBuf,
    overwrite: bool,
) -> Result<PathBuf, String> {
    utils::fs::compress_tarball(src_dir, dest_file, overwrite).map_err(|e| e.to_string())
}
